yarn-project/prover-client/src/orchestrator/orchestrator_workflow.test.ts

```diff
@@ -87,19 +82,18 @@ describe('prover/orchestrator', () => {
         orchestrator.startNewEpoch(1, 1, 1);
         await orchestrator.startNewBlock(globalVariables, [message], previousBlockHeader);
 
-        await sleep(10);
-        expect(mockProver.getBaseParityProof).toHaveBeenCalledTimes(NUM_BASE_PARITY_PER_ROOT_PARITY);
-        expect(mockProver.getRootParityProof).not.toHaveBeenCalled();
-
-        await sleep(10);
-        // even now the root parity should not have been called
+        // the prover broker deduplicates jobs, so the base parity proof
+        // for the three sets empty messages is called only once. so total
+        // calls is one for the empty messages and one for the custom message.
+        await sleep(2000);
+        expect(mockProver.getBaseParityProof).toHaveBeenCalledTimes(2);
         expect(mockProver.getRootParityProof).not.toHaveBeenCalled();
 
         // only after the base parity proof is resolved, the root parity should be called
         pendingBaseParityResult.resolve(expectedBaseParityResult);
 
         // give the orchestrator a chance to calls its callbacks
-        await sleep(10);
+        await sleep(5000);
```

Comment:
We really need to move away from sleep and use fake timers 🥹 but this looks good and it gets us going!